api/handlers/2fa.go

```diff
@@ -21,6 +21,20 @@ func TwofactorHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Validate the token
+	authHeader := r.Header.Get("Authorization")
+	matches := AuthBearerRegexp.FindStringSubmatch(authHeader)
+	if len(matches) == 0 {
+		http.Error(w, "No authorization token header found", http.StatusInternalServerError)
+		return
+	}
+
+	jwtToken := matches[1]
+	if valid, err := account.ValidJwtToken(jwtToken, model.BasicAuthAudience); !valid {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
```

Comment:
Perhaps consider moving this into a function?  Here, starting at line 59, and line 135 all have the same code.

Reply:
👍 can do!